pilot/pkg/leaderelection/leaderelection.go

```diff
@@ -136,3 +142,25 @@ func NewLeaderElection(namespace, name, electionID string, client kubernetes.Int
 		cycle: atomic.NewInt32(0),
 	}
 }
+
+// NewPerRevisionLeaderElection calls a leader election scoped to the replicas of a given revision.
+// For instance, in webhook patching, we want each revision to handle its own patching, but only a
+// single replica within that revision.
+func NewPerRevisionLeaderElection(namespace, name, revision, electionID string, client kubernetes.Interface) *LeaderElection {
+	if name == "" {
+		name = "unknown"
+	}
+	if revision != "" {
+		electionID = fmt.Sprintf("%s-%s", electionID, revision)
+	}
+	return &LeaderElection{
```

Comment:
nit: make this call `NewLeaderElection` and just have the function modify the `electionID` arg

Reply:
Done.